src/main/kotlin/pl/edu/uj/ii/ksi/mordor/services/repository/RepositoryService.kt

```diff
@@ -11,19 +12,15 @@ import java.nio.file.Paths
 @Service
 class RepositoryService(@Value("\${mordor.root_path}") private val rootPathStr: String) {
     private final val rootPath: Path = Paths.get(rootPathStr)
+    private final val logger = LoggerFactory.getLogger(RepositoryService::class.java)
 
     private fun getRequestedPath(path: String): Path {
-        val parts = path.split("/")
-        var current = rootPath
-        for (i in 2..(parts.size - 1)) {
```

Comment:
The old code was ignoring the first part of the url (`/files/` or `/download/`). It should be striped by the controller (so only the relative path is passed to `getRequestedPath`), but it seems that I haven't noticed this when I moved this method from the controlled to this service. Could you fix the controller as well?

Reply:
Nevermind, fixed in 61e76412e7f01dbcaaa02de3a5fb10b9ae553f64